library/src/updater.rs

```diff
@@ -128,6 +130,20 @@ fn libapp_path_from_settings(original_libapp_paths: &[String]) -> Result<PathBuf
     first.map(PathBuf::from)
 }
 
+pub fn with_state<F, R>(f: F) -> anyhow::Result<R>
```

Comment:
This is always mut.  Some callers don't need/want the mut version.  I wonder if there should be a with_state and with_mut_state?

Reply:
Split these out